client/src/components/generic/Table.js

```diff
@@ -262,8 +263,8 @@ export const Table = ({
                 <Checkbox
                   color='primary'
                   disabled={isLoading || rowsCount === 0}
-                  indeterminate={selectedRows.length > 0 && selectedRows.length < rowsCount}
-                  checked={rowsCount > 0 && selectedRows.length === rowsCount}
+                  indeterminate={selectedRows.length > 0 && selectedRows.length < rows.length}
+                  checked={rowsCount > 0 && selectedRows.length === rows.length}
```

Comment:
We probably can completely replace `rowsCount` with `rows.length`; and the remove it

Reply:
I renamed rowsCount to totalRowsCount (it's used for pagination, can't get rid of it completely) and made a new variable rowsOnPage = rows.length.